docs/development/quickstart.rst

```diff
@@ -0,0 +1,365 @@
+
+    git checkout main
+    git pull upstream main --ff-only
+
+Now create a feature branch for making your changes. For example::
+
+    git switch -c subpackage-new-feature
+
+This changes your working branch from ``main`` to the ``subpackage-new-feature`` branch.
+Keep any changes in this branch specific to one bug or feature so it is clear what the
+branch brings to astropy. You can have many feature branches and switch in between them
+using the ``git switch`` command.
+
+Using a descriptive branch name can help you stay organized. For example
+```io-ascii-commented-header-15513``` might be a good name for a branch that fixes the
+commented header issue `#15513 <https://github.com/astropy/astropy/issues/15513>`_ in
+the ``io.ascii`` sub-package.
+
+.. Important:: Never use the ``main`` branch for making changes. Always create a new
+   feature branch for your changes.
+
+When you want to update the feature branch with changes in main after
+you created the branch, check the section on
+:ref:`updating a PR <contributing.update-pr>`.
+
+.. _contributing.commit-code:
+
+Making code or documentation changes
+------------------------------------
+
+Now comes the fun part where you use your favorite editor or IDE to make changes to the
+code or documentation! At a high level this breaks into a few parts:
+
+- **Make changes**: Make the changes you want to make. This could be fixing a bug,
+  adding a new feature, or updating the documentation.
+- **Test changes**: For code changes, ensure that they work as expected following the
+  process outlined in the :ref:`testing-guidelines` section.
+- **Build documentation**: If you are updating the documentation, you will want to
+  :ref:`build the documentation <builddocs>` to ensure that it looks good.
+- **Add a changelog entry**: For most code changes you will need to
+  :ref:`add-changelog`.
+
+.. tip:: For more information and examples see :ref:`edit-flow` section.
+
+You can see all the changes you've currently made by running:
+
+.. code-block:: shell
+
+    git status
+
+You can then commit your all your changes to your local repository with an explanatory
+commit message:
+
+.. code-block:: shell
+
+    git add files-that-you-changed ...
+    git commit -m "your commit message goes here"
+
+You should make frequent commits and always include a commit message. Each commit
+should represent one logical set of changes.
+
+.. Important:: Never merge changes from ``upstream/main`` into your feature branch. If
+   changes in ``main`` require changes to our code you must :ref:`rebase`.
+
+.. _contributing.push-code:
+
+Pushing your changes
+--------------------
+
+When you want your changes to appear publicly on your GitHub page, push your
+forked feature branch's commits::
+
+    git push origin subpackage-new-feature
+
+Here ``origin`` is the default name given to your remote repository on GitHub.
+
+
+
+You can see the remote repositories
```

Comment:
This might be better at the end of the part where `upstream` is added.

Reply:
Done